### ISSUE TITLE
Assertion error for binary cross-compiled with gcc-11.1 for aarch64

### ISSUE THREAD

**Issue by YashasAndaluri** [eliben/pyelftools#412]
Hello
I’m trying to use angr to analyze a binary from findutils that has been built (with `-g`) on x86 with a gcc-11.1 cross-compiler for the aarch64-linux-gnu target. However, I encounter the following assertion error with pyelftools.

```
File "$HOME/angr.py", line 128, in <module>
proj = angr.Project(args.binary ,auto_load_libs=False, load_debug_info=True, arch='aarch64')
File "$HOME/miniconda3/envs/angr/lib/python3.6/site-packages/angr/project.py", line 135, in __init__
self.loader = cle.Loader(self.filename, concrete_target=concrete_target, **load_options)
 File "$HOME/miniconda3/envs/angr/lib/python3.6/site-packages/cle/loader.py", line 133, in __init__
    self.initial_load_objects = self._internal_load(main_binary, *preload_libs, *force_load_libs, preloading=(main_binary, *preload_libs))
File "$HOME/miniconda3/envs/angr/lib/python3.6/site-packages/cle/loader.py", line 668, in _internal_load
    obj = self._load_object_isolated(main_spec)
File "$HOME/miniconda3/envs/angr/lib/python3.6/site-packages/cle/loader.py", line 850, in _load_object_isolated
    result = backend_cls(binary, binary_stream, is_main_bin=self.main_object is None, loader=self, **options)
File "$HOME/miniconda3/envs/angr/lib/python3.6/site-packages/cle/backends/elf/elf.py", line 168, in __init__
    self._load_dies(dwarf)
File "$HOME/miniconda3/envs/angr/lib/python3.6/site-packages/cle/backends/elf/elf.py", line 623, in _load_dies
    for cu in dwarf.iter_CUs():
File "$HOME/miniconda3/envs/angr/lib/python3.6/site-packages/elftools/dwarf/dwarfinfo.py", line 353, in _parse_CUs_iter
    cu = self._cached_CU_at_offset(offset)
File "$HOME/miniconda3/envs/angr/lib/python3.6/site-packages/elftools/dwarf/dwarfinfo.py", line 383, in _cached_CU_at_offset
    cu = self._parse_CU_at_offset(offset)
File "$HOME/miniconda3/envs/angr/lib/python3.6/site-packages/elftools/dwarf/dwarfinfo.py", line 421, in _parse_CU_at_offset
    dwarf_version=cu_header['version'])
File "$HOME/miniconda3/envs/angr/lib/python3.6/site-packages/elftools/dwarf/structs.py", line 93, in __init__
    assert address_size == 8 or address_size == 4
AssertionError
```

The address_size here is 0.
Any insights on this error would greatly help.

**Comment by sevaa** [comment]
Can you share the binary? If not, can you build a smaller binary ("Hello world") with the same compiler and settings that would exhibit a similar error, and share that?

**Comment by YashasAndaluri** [comment]
Binary that resulted in the above error: [find.zip](https://github.com/eliben/pyelftools/files/8838670/find.zip)

**Comment by sevaa** [comment]
The binary fails with the *published* version of pyelftools, but works with the current master. It's a known issue that's been patched in PR #400.

Until the next release of pyelftools, and then some time for the maintainers of angr to integrate that, may I suggest that you apply the following monkey patch for pyelftools first thing in your program: https://github.com/sevaa/dwex/blob/master/dwex/patch.py

    from patch import monkeypatch
    monkeypatch()

It repeats some of the recent unreleased fixes to pyelftools. The very first patched method (`_create_abbrev_declaration_ex`) should fix your exception.
